substate-store-impls/src/hash_tree/partition_tier.rs

```diff
@@ -108,8 +112,8 @@ impl<'s, S: ReadableTreeStore + WriteableTreeStore> PartitionTier<'s, S> {
                     });
                     (partition, new_leaf)
                 });
-        let (new_root_hash, _) =
-            self.apply_leaf_updates(BaseTree::Existing, next_version, leaf_updates);
-        new_root_hash
+        let update_batch = self.generate_tier_update_batch(next_version, leaf_updates);
+        self.apply_tier_update_batch(&update_batch);
```

Comment:
on the other hand, when I moved the `apply()` call as the last action inside `generate()`, it becomes even more "streamlined".
(ofc the `generate_and_apply()` still returns `new_root_hash+tree_update_batch`, since callers [selectively] want them)

Does the `generate() | apply()` split have some other advantages I don't see?

Reply:
As for whether `next_version` is in the update batch, I don't feel strongly.

I was thinking that it's nice if the update batch captures all the relevant results and updates from running the generation process; and can be used for both commit or other things we need to do (e.g. return the root hash or attach our substate values).

But as I say, I don't feel strongly

Reply:
I would feel more strongly if it was some more "public" API, but here I can live with this 👍